src/c/types/array.c

```diff
@@ -43,24 +52,21 @@ bool ucdr_serialize_array_byte_2(ucdrBuffer* ub, const ucdrEndianness endianness
     uint32_t array_size = size * data_size;
     uint32_t alignment = ucdr_buffer_alignment(ub, sizeof(uint16_t));
 
-    if(ucdr_check_buffer(ub, alignment + array_size))
-    {
-        ub->iterator += alignment;
-        if(UCDR_MACHINE_ENDIANNESS == endianness)
-        {
-            memcpy(ub->iterator, array, array_size);
+    ub->iterator += alignment;
```

Comment:
The alignment ALWAYS fit if the buffer is multiply of 8 (Put it as a mandatory requirement?).
If the buffer is not 8 alignment, is not a problem because an error=true will be returned and the ucdrBuffer.error will set properly.

Reply:
Outdated